restart_process/README.md

```diff
@@ -1,10 +1,101 @@
+docker_build_with_restart(
+    'foo-image',
+    './foo',
+    '/go/bin/foo',
+    arg1=val1,
+    arg2=val2,
+    live_update=[x, y, z...]
+)
+```
+The call above looks just like the initial `docker_build` call except for one added parameter, `entrypoint` (in this example, `/go/bin/foo`). This is the command that you want to run on container start and _re-run_ on Live Update.
+
+## What's Happening Under the Hood
+
+This extension makes use of [`entr`](https://github.com/eradman/entr/), a utility for running arbitrary commands when files change. Specifically, we override the container's entrypoint with the following:
+
+```python
+echo '/.restart-proc' | entr -rz <entrypoint>
+```
+
+This invocation says:
+- when the container starts, run <entrypoint>
+- whenever the `/.restart-proc` file changes, re-execute <entrypoint>
+- `-z`: if <entrypoint> exits, `entr` should exit as well (so that k8s/Tilt can detect that the container has stopped)[¹](#fn1).
+
+We also set the following as the last `live_update` step:
+```python
+run('date > /.restart-proc')
+```
+
+Because `entr` will re-execute the entrypoint whenever `/.restart-proc'` changes, the above `run` step will cause the entrypoint to re-run.
+
+#### Provide `entr`
+For this all to work, the `entr` binary must be available on the Docker image. The easiest solution would be to call e.g. `apt-get install entr` in the Dockerfile, but different base images will have different package mangers; rather than grapple with that, we've made a statically linked binary available on Docker image: [`tiltdev/entr`](https://hub.docker.com/repository/docker/tiltdev/entr).
+
+If you want to build `image-foo`, this extension will:
+- build your requested image as `image-foo-base` via `docker_build` call with all of your specified args/kwargs
+- build `image-foo` (the actual image that will be used in your resource) as a _child_ of `image-foo-base`:
+```Dockerfile
+FROM tiltdev/entr:2020-16-04 as entr-img
+
+FROM image-foo-base
+
+# we'll use this file to signal restarts,
+# make sure it exists
+RUN ["touch", "/.restart-proc"]
+
+# make the entr binary available on this image
+COPY --from=entr-img /entr /
+```
+
+Thus, the final image produced is tagged `image-foo` and has all the properties of your original `docker_build`, plus access to the `entr` binary.
+
+## Unsupported Cases
+This extension does NOT support process restarts for:
+- Docker images without shell support (e.g. `scratch`)
+- Images run in Docker Compose resources
+- `custom_build`
+- ???
+
+Run into a bug? Need a use case that we don't yet support? Let us know--[open an issue](https://github.com/windmilleng/tilt-extensions/issues) or [contact us](https://tilt.dev/contact).
+
+## For Maintainers: Releasing
+TODO(maia) 👀
 
-TODO: documentation
-TODO: release instructions
-TODO: custom_build support
+---
+### Notes
+<a name="fn1"></a>**1**: the `-z` flag is only available in `entr` >= [v4.5](https://github.com/eradman/entr/releases/tag/4.5)
```

Comment:
I'm not sure this will be very interesting to readers of the doc, since the doc doesn't describe any workflows in which users might be usign entr < 4.5? But w/e.